### PR TITLE
bp_is_root_blog check added on bp_core_menu_highlight_parent_page function

### PR DESCRIPTION
**Pull request by robinwpdeveloper**
bp_is_root_blog check added on bp_core_menu_highlight_parent_page function

Trac ticket: [7308](https://buddypress.trac.wordpress.org/ticket/7308)

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

**Comment by robinwpdeveloper** [comment]
@imath can you please review this pull request.
This is my first contribution to buddypress :)

**Comment by imath** [comment]
@robinwpdeveloper Sure & with great pleasure! Thanks a lot for this first PR 😍. Don't hesitate to pull more requests 👌

File: src/bp-core/bp-core-filters.php [hunk 1]
```diff
@@ -168,7 +168,7 @@ function bp_core_exclude_pages_from_nav_menu_admin( $object = null ) {
  * @return array
  */
 function bp_core_menu_highlight_parent_page( $retval, $page ) {
-	if ( ! is_buddypress() ) {
+	if ( ! is_buddypress() || bp_is_root_blog() ) {
```
Comment (imath) [inline src/bp-core/bp-core-filters.php:171]: If we leave this condition this way, it means we don't filter css classes for listed pages if we're not in a BuddyPress area or if it's the WordPress site BuddyPress is activated on. So the `current_page_item` class is never added to the `li` item of the current page when it's actually the case. You can test your patch going on any member's page (eg: site.url/members/anyone) , the class will be missing.

The right check should be, we will not filter the css classes if it's not a BuddyPress area or if we are **not** viewing the WordPress site BuddyPress is activated on. Can you review the PR and add a `!` before `bp_is_root_blog()` to meet this need?

```suggestion
	if ( ! is_buddypress() || ! bp_is_root_blog() ) {
```

**Review by imath — CHANGES_REQUESTED** [review]
Thanks again for your PR, could you do the `!` minor change so I can commit it?

PS: so that you can improve your next PRs, I advise you to test your code changes 😉 .

**Comment by imath** [comment]
FYI, a patch was added to our [Trac ticket](https://buddypress.trac.wordpress.org/ticket/7308) about it, so I'm going to close this PR. I hope you won't be disappointed, no worries about the credit/prop, I'll add your WordPress username to my commit.

**Pull request closed without merging**